### PR TITLE
Use proper connector session in ConnectorNodePartitioningProvider

### PR DESCRIPTION
**Pull request by Crossoverrr**
Fix cannot fetch session properties in method `ConnectorNodePartitioningProvider#getSplitBucketFunction`

### CODE REVIEW

**Review by sopel39 — APPROVED** [review]

File: presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java [hunk 1]
```diff
@@ -221,7 +221,7 @@ private ConnectorBucketNodeMap getConnectorBucketNodeMap(Session session, Partit
 
         ToIntFunction<ConnectorSplit> splitBucketFunction = partitioningProvider.getSplitBucketFunction(
                 partitioningHandle.getTransactionHandle().orElse(null),
-                session.toConnectorSession(),
+                session.toConnectorSession(partitioningHandle.getConnectorId().get()),
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java:224]: There are many places where `session.toConnectorSession()` is called, and only one is updated.

Would it make sense to have a test ensuring at least the "connector partitioning interface" use-cases is working end to end?

Comment (Crossoverrr) [reply presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java:224]: The title is confusing, I just test a method `ConnectorNodePartitioningProvider#getSplitBucketFunction`, it cannot fetch the session properties.

Comment (sopel39) [reply presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java:224]: Could you add a test for that?

Comment (Crossoverrr) [reply presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java:224]: Screenshot from 2020-10-26 17-49-51
`NodePartitioningManager#getConnectorBucketNodeMap` has the same problem as `ConnectorNodePartitioningProvider#getSplitBucketFunction`.

Comment (Crossoverrr) [reply presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java:224]: Well, I saw method `Session#toConnectorSession` has been test in many place in presto test code, do we really need a test for that?   :)

Comment (sopel39) [reply presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java:224]: Please also fix `NodePartitioningManager#getConnectorBucketNodeMap`

Comment (Crossoverrr) [reply presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java:224]: Sure!

File: presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java [hunk 2]
```diff
@@ -207,7 +207,7 @@ private ConnectorBucketNodeMap getConnectorBucketNodeMap(Session session, Partit
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java:207]: change commit title: `Use proper connector session in ConnectorNodePartitioningProvider`

**Review by sopel39 — APPROVED** [review]

File: presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java (hunk 2 above)
Comment (Crossoverrr) [reply presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java:207]: Title changed!   :)

File: presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java (hunk 1 above)
Comment (findepi) [reply presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java:224]: Is it possible `partitioningHandle.getConnectorId().isEmpty()` here?

Comment (Crossoverrr) [reply presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java:224]: I think it won't be null, for it is checked in the previous code.

**Comment by sopel39** [comment]
merged, thanks!

**Pull request merged**